lua/lualine.lua

```diff
@@ -293,16 +274,45 @@ local function statusline(is_focused)
   return table.concat(status)
 end
 
+-- check if any extension matches the filetype and return proper sections
+local function get_extension_sections()
+  local sections, inactive_sections = nil, nil
+  for _, extension in ipairs(M.extensions) do
+    for _, filetype in ipairs(extension.filetypes) do
+      if vim.bo.filetype == filetype then
+        sections = extension.sections
+        inactive_sections = extension.inactive_sections
+        break
+      end
+    end
+  end
+  return {sections = sections, inactive_sections = inactive_sections}
+end
+
 local function status_dispatch()
+  local extension_sections = get_extension_sections()
   if vim.g.statusline_winid == vim.fn.win_getid() then
-    return statusline(true)
+    local sections = extension_sections.sections
+    if sections == nil then
+      sections = M.sections
+    end
+    return statusline(sections, true)
   else
-    return statusline(false)
+    local inactive_sections = extension_sections.inactive_sections
+    if inactive_sections == nil then
+      inactive_sections = M.inactive_sections
+    end
+    return statusline(inactive_sections, false)
   end
 end
 
 local function tabline()
-  return statusline(M.tabline, true)
+  local extension_sections = get_extension_sections()
```

Comment:
This doesn't make sense here . If an extention is used itwill apear both on statusline and tabline . It will conflict with 3rd party tablines and default tabline ( tabline was suppose to be purely o[t in feature) . Also it's weird if same thing apearing both at top and bottom . Also I was wondering if by your defination of extention will they ever need complexcity of separate config fro inactive an active section ? None of the examples do . We can probably just show same section for active and inactive state . Otherwize lgtm :).

Reply:
you are right, they can be simplified